src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -1115,6 +1126,23 @@ impl<'tcx> LocationTree {
                 })
             },
         )?;
+        // For a wildcard access to be valid each subtree needs to either contain an exposed tag
+        // through which the access could have happened or a foreign access to the subtree must
+        // be possible. If neither of these is the case than the access is UB.
+        // In reality this is only ever UB on the main subtree as all other trees always allow
+        // foreign accesses.
+        if !has_exposed
+            // Check that no access that is foreign to this subtree is possible.
+            // (Its only impossible for the main subtree).
```

Comment:
```suggestion
        // If there is no exposed node in this tree that allows this access, then the
        // access *must* be foreign. So we check if the root of this tree would allow this
        // as a foreign access, and if not, then we can error.
        // In practice, all wildcard trees accept foreign accesses, but the main tree does
        // not, so this catches UB when none of the nodes in the main tree allows this access.
        if !has_exposed
```